drgn/helpers/linux/rss.py

```diff
@@ -0,0 +1,148 @@
+# Copyright (c) 2025 Oracle and/or its affiliates
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+
+"""
+The ``drgn.helpers.linux.rss`` module provides get_task_rss helper to get resident set size
+in pages of a given task .
+"""
+
+from typing import Dict, Iterable, NamedTuple, Optional
+
+from drgn import Object, Program, TypeKind
+from drgn.helpers.common.prog import takes_program_or_default
+from drgn.helpers.linux.list import list_for_each_entry
+from drgn.helpers.linux.percpu import percpu_counter_sum
+
+__all__ = ("get_task_rss",)
+
+
+class TaskRss(NamedTuple):
+    """
+    Represent's a task's resident set size in pages. See task_rss().
+    """
+
+    rss_file: int
+    rss_anon: int
+    rss_shmem: int
+    swapents: int
+
+    @property
+    def total(self) -> int:
+        return self.rss_file + self.rss_anon + self.rss_shmem
+
+
+def for_each_task_in_group(
+    task: Object, include_self: bool = False
+) -> Iterable[Object]:
+    """
+    Iterate over all tasks in the thread group
+
+    Or, in the more common userspace terms, iterate over all threads of a
+    process.
+
+    :param task: a task whose group to iterate over
+    :param include_self: should ``task`` itself be returned
+    :returns: an iterable of every thread in the thread group
+    """
+    if include_self:
+        yield task
+    if hasattr(task, "thread_group"):
+        yield from list_for_each_entry(
+            "struct task_struct",
+            task.thread_group.address_of_(),
+            "thread_group",
+        )
+    else:
+        # Since commit 8e1f385104ac0 ("kill task_struct->thread_group") from
+        # 6.7, the thread_group list is gone, replaced by a list inside the
+        # task.signal struct. This has an explicit list_head (unlike the
+        # thread_group which just linked each task together with no explicit
+        # head node).
+        for other in list_for_each_entry(
+            "struct task_struct",
+            task.signal.thread_head.address_of_(),
+            "thread_node",
+        ):
+            # We've already yielded "task" (or not, depending on the caller's
+            # preference) so skip it here.
+            if other != task:
+                yield other
+
+
+@takes_program_or_default
+def get_task_rss(
```

Comment:
It looks like the closest thing the kernel has to this helper is the `get_mm_rss()` function: https://github.com/torvalds/linux/blob/master/include/linux/mm.h#L2609. Maybe we can rename this to `get_task_rss_info()` or something like that to indicate that it returns a tuple instead of just a single counter? That leaves the possibility open for adding `get_mm_rss()`, `get_mm_rss_info()`.